### PR TITLE
bigtable.version 1.14.0 and tweaked dependency declarations for RequireUpperBoundDeps

### LINKED ISSUE #2604: Bump bigtable.version to 1.14.0
**Is your feature request related to a problem? Please describe.**
Apache Beam [depends on](https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L483) bigtable-client-core. We plan to upgrade gax library of Beam to 1.57.1 to support new features of other client library, howerver, the bigtable-client-core still uses old version, which caused version compatibility issue. Upgrading bigtable-client-core's [bigtable.version](https://github.com/googleapis/java-bigtable-hbase/blob/master/pom.xml#L59)  to 1.14.0 would solve the problem.

**Describe the solution you'd like**
Upgrading bigtable-client-core's [bigtable.version](https://github.com/googleapis/java-bigtable-hbase/blob/master/pom.xml#L59)  to 1.14.0

**Describe alternatives you've considered**
No alternative.

**Additional context**
 I've tried updating the version locally to 1.14.0, but build failed due to 3 dependency issues. It looks like a non-trivial version bump.

======
[WARNING] Rule 2: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for io.grpc:grpc-protobuf-lite:1.29.0 paths to dependency are:

+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-com.google.api.grpc:grpc-google-common-protos:1.18.0
        +-io.grpc:grpc-protobuf-lite:1.29.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-io.grpc:grpc-protobuf:1.30.2
        +-io.grpc:grpc-protobuf-lite:1.30.2

Require upper bound dependencies error for com.google.http-client:google-http-client-jackson2:1.35.0 paths to dependency are:
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud:google-cloud-core:1.93.7
      +-com.google.http-client:google-http-client-jackson2:1.35.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase:1.14.1-SNAPSHOT
    +-com.google.auth:google-auth-library-oauth2-http:0.21.1
      +-com.google.http-client:google-http-client-jackson2:1.35.0 (managed) <-- com.google.http-client:google-http-client-jackson2:1.36.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-com.google.cloud:google-cloud-core-http:1.93.7
        +-com.google.api-client:google-api-client:1.30.9
          +-com.google.http-client:google-http-client-jackson2:1.35.0 (managed) <-- com.google.http-client:google-http-client-jackson2:1.34.2
, 

Require upper bound dependencies error for io.grpc:grpc-core:1.30.0 paths to dependency are:
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-io.grpc:grpc-core:1.30.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase:1.14.1-SNAPSHOT
    +-io.grpc:grpc-core:1.30.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-io.grpc:grpc-core:1.30.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-io.grpc:grpc-core:1.30.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud:google-cloud-bigtable:1.14.0
      +-io.grpc:grpc-core:1.30.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-benchmarks:1.14.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x:1.14.1-SNAPSHOT
    +-com.google.cloud.bigtable:bigtable-client-core:1.14.1-SNAPSHOT
      +-io.grpc:grpc-grpclb:1.30.2
        +-io.grpc:grpc-core:1.30.2

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:36 min
[INFO] Finished at: 2020-08-03T11:03:48-07:00
[INFO] ------------------------------------------------------------------------

### PR DESCRIPTION
**Pull request by suztomo**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #2604 ☕️

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/pom.xml [hunk 1]
```diff
@@ -54,6 +54,13 @@ limitations under the License.
     <dependency>
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-bigtable</artifactId>
+      <exclusions>
+        <exclusion>
+          <!-- We declare grpc-core dependency -->
+          <groupId>io.grpc</groupId>
+          <artifactId>grpc-core</artifactId>
+        </exclusion>
```
Comment (suztomo) [inline bigtable-client-core-parent/bigtable-client-core/pom.xml:62]: This explicit exclusion passes RequireUpperBoundDeps check.

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/pom.xml [hunk 2]
```diff
@@ -9,23 +9,6 @@
 
   <artifactId>bigtable-hbase-1.x-benchmarks</artifactId>
 
-  <!--  TODO: remove this dependency when upgraded transitively to 1.34.2 -->
```
Comment (suztomo) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/pom.xml:12]: https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/0.8.3/pom  has http-client version 1.36.0.

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/pom.xml [hunk 3]
```diff
@@ -9,23 +9,6 @@
...
-  <dependencyManagement>
-    <dependencies>
-      <dependency>
-        <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client-jackson2</artifactId>
-        <version>1.35.0</version>
-      </dependency>
-      <!--  todo(kolea2): temporary workaround until api-common and google-cloud-core are updated to 1.7.1-->
```
Comment (suztomo) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/pom.xml:20]: @kolea2 https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/0.8.3/pom has api-common 1.9.3 and google-cloud-core 1.93.7.

File: bigtable-client-core-parent/bigtable-client-core/pom.xml [hunk 4]
```diff
@@ -64,6 +71,13 @@ limitations under the License.
     <dependency>
       <groupId>com.google.api.grpc</groupId>
       <artifactId>grpc-google-common-protos</artifactId>
+      <exclusions>
+        <exclusion>
+          <!-- We use grpc-protobuf-lite from grpc-protobuf -->
```
Comment (suztomo) [inline bigtable-client-core-parent/bigtable-client-core/pom.xml:76]: grpc-protobuf is at line 171 in this file.

**Comment by kolea2** [comment]
@suztomo thanks for the PR! As an FYI, the master branch is currently working on a 2.0.0 release - we are maintaining the 1.x branch here: https://github.com/googleapis/java-bigtable-hbase/tree/bigtable-1.x and it has already been bumped to 1.14.0. Once https://github.com/googleapis/java-bigtable-hbase/pull/2592 is merged, we will release this as 1.16.0.

**Comment by suztomo** [comment]
@kolea2 Thank you for response. Closing this PR as the branch https://github.com/googleapis/java-bigtable-hbase/tree/bigtable-1.x already has 1.14.0.

**Comment by terryxian78** [comment]
@kolea2 Thanks, very eager to see 1.16.0 release with bigtable.version bumped. Do you know when it would be released?

**Pull request closed without merging**